src/tiled/editpolygontool.cpp

```diff
@@ -566,6 +596,59 @@ void EditPolygonTool::updateMovingItems(const QPointF &pos,
     }
 }
 
+void EditPolygonTool::addNode(const QPointF &pos,
```

Comment:
The name `addNode` does not seem to reflect what the function is doing. I think `updateAddingNode` would be better, but it still leaves two kinds of terminology for the same thing. Since you're already using the mode `Extending` and have `mExtendingFirst`, I would suggest to rename all the other functions as follows:
* `updateExtending`
* `finishExtending`
* `cancelExtending`